### PR TITLE
feat: last message markdown [WPB-7413]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7413" title="WPB-7413" target="_blank">TaskWPB-7413</a>  [Android] Make drafts visible on conversation list
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- implemented markdown in `LastMessageSubtitle`
- UI improvements in code block
- option to disable links and mentions actions in `inlineNodeChildren`
- implemented MarkdownParser to have single source of true for parsing markdowns
- removed unused date variable from MessagePreview 
- removed unnecessary `MessageComposition.DEFAULT` and moved default values to constructor
- added new `MessagePreviewContent.Draft`
- added `EMPTY_SPACE = "&nbsp;"` to handle properly empty space for MarkdownInline

markdown

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [4211](https://10.10.124.134/job/AR-build-pipeline/4211/) **failed**.

**Comment by AndroidBob** [comment]
Build [4215](https://10.10.124.134/job/AR-build-pipeline/4215/) **failed**.

**Comment by AndroidBob** [comment]
Build [4216](https://10.10.124.134/job/AR-build-pipeline/4216/) **failed**.

**Comment by AndroidBob** [comment]
Build [4217](https://10.10.124.134/job/AR-build-pipeline/4217/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9422472)(9422472)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9422472)(9422472)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/markdown/MarkdownConstants.kt [hunk 1]
```diff
@@ -17,9 +17,19 @@
...
+import org.commonmark.Extension
+import org.commonmark.ext.gfm.strikethrough.StrikethroughExtension
+import org.commonmark.ext.gfm.tables.TablesExtension
+
 object MarkdownConstants {
     const val TAG_URL = "linkTag"
     const val TAG_MENTION = "mentionTag"
     const val MENTION_MARK = "&&"
     const val BULLET_MARK = "\u2022"
+    const val EMPTY_SPACE = "&nbsp;"
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/markdown/MarkdownConstants.kt]: suggestion: change name to `NONBREAKING_SPACE` to avoid confusion

**Review by saleniuk — APPROVED** [review]

**Review by yamilmedina — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4229](https://10.10.124.134/job/AR-build-pipeline/4229/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9429794)(9429794)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9429794)(9429794)-staging-compat.apk)

**Pull request merged**